tests/e2e-pw/specs/merchant/merchant-orders-partial-refund.spec.ts

```diff
@@ -0,0 +1,178 @@
+ */
+import { test, expect, Page } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { config } from '../../config/default';
+import { getMerchant, getShopper } from '../../utils/helpers';
+import {
+	fillCardDetails,
+	placeOrder,
+	setupProductCheckout,
+} from '../../utils/shopper';
+import { goToShop } from '../../utils/shopper-navigation';
+import { goToOrder } from '../../utils/merchant-navigation';
+import {
+	activateMulticurrency,
+	deactivateMulticurrency,
+	restoreCurrencies,
+} from '../../utils/merchant';
+
+// Needs to be finished.
+test.describe( 'Order > Partial refund', () => {
+	const product1 = config.products.simple.name;
+	const product2 = 'Belt';
+	const product3 = 'Hoodie with Logo';
+
+	/**
+	 * Elements:
+	 * - test title
+	 * - object containing the items to be ordered, and the quantities and amounts to be refunded
+	 */
+	const dataTable: [
+		string,
+		{
+			lineItems: Array< [ string, number ] >;
+			refundInputs: { refundQty: number; refundAmount: number }[];
+		}
+	][] = [
+		[
+			'Partially refund one product of two product order',
+			{
+				lineItems: [
+					[ product1, 1 ],
+					[ product2, 1 ],
+				],
+				refundInputs: [ { refundQty: 0, refundAmount: 5 } ],
+			},
+		],
+		[
+			'Refund two products of three product order',
+			{
+				lineItems: [
+					[ product1, 1 ],
+					[ product2, 2 ],
+					[ product3, 1 ],
+				],
+				refundInputs: [
+					{ refundQty: 1, refundAmount: 18 },
+					{ refundQty: 1, refundAmount: 55 },
+				],
+			},
+		],
+	];
+
+	let firstOrderId: string;
+	let secondOrderId: string;
+	let orderIds: string[];
+	let orderTotal: string;
+	let wasMulticurrencyEnabled = false;
+	let merchantPage: Page, shopperPage: Page;
+
+	const orderProducts = async ( { browser }, dataTableIndex: number ) => {
+		shopperPage = ( await getShopper( browser ) ).shopperPage;
+		const lineItems = dataTable[ dataTableIndex ][ 1 ].lineItems;
+		await goToShop( shopperPage );
+		await setupProductCheckout( shopperPage, lineItems );
+		await fillCardDetails( shopperPage );
+		await placeOrder( shopperPage );
+		await expect(
+			shopperPage.getByRole( 'heading', { name: 'Order received' } )
+		).toBeVisible();
+
+		const orderIdField = shopperPage.locator(
+			'.woocommerce-order-overview__order.order > strong'
+		);
+		const orderId = await orderIdField.innerText();
+
+		return orderId;
+	};
+
+	test.beforeAll( async ( { browser } ) => {
+		merchantPage = ( await getMerchant( browser ) ).merchantPage;
+		wasMulticurrencyEnabled = await activateMulticurrency( merchantPage );
+		await restoreCurrencies( merchantPage );
+		firstOrderId = await orderProducts( { browser }, 0 );
+		secondOrderId = await orderProducts( { browser }, 1 );
+		orderIds = [ firstOrderId, secondOrderId ];
+	} );
+
+	test.afterAll( async () => {
+		if ( ! wasMulticurrencyEnabled ) {
+			await deactivateMulticurrency( merchantPage );
+		}
+	} );
+
+	for ( let i = 0; i < dataTable.length; i++ ) {
```

Comment:
Nitpick: You can use this for accessing the elements in the `dataTable` easily:

```js
Object.entries(dataTable).forEach(([title, { lineItems, refundInputs }]) => {
```

Reply:
Thanks for this suggestion. I wasn't able to implement it exactly this way. I also need the `i` for targeting the correct `orderId` from the `orderIds` array. I took a similar approach in [f68aa46](https://github.com/Automattic/woocommerce-payments/pull/10194/commits/f68aa46743a0488076b35bd496fce9e22b683d7f).

Reply:
You could get the index with:

```js
Object.entries(dataTable).forEach(([title, { lineItems, refundInputs }], i) => {
```

Reply:
Yea, I got the index with the `i` like that, but Typescript doesn't like when I use `Object.entries(dataTable).forEach` instead of just `dataTable.forEach`. If I use your suggestion, I get:

```
Property 'refundInputs' does not exist on type '[string, { lineItems: [string, number][]; refundInputs: { refundQty: number; refundAmount: number; }[]; }]'.ts(2339)
```

I don't get it with `lineItems` b/c I remove them since they aren't used in the test, only in the beforeAll setup. If I add them back, I also get the Typescript warning about them not existing on the dataTable type.

`dataTable.forEach` is working well though, so I think we can stick with it.